javaslang/src/test/java/javaslang/collection/BitSetTest.java

```diff
@@ -0,0 +1,389 @@
+        Mapper<T> mapper = new Mapper<>();
+        return BitSet.withRelations(mapper::fromInt, mapper::toInt);
+    }
+
+    @Override
+    protected <T> Collector<T, ArrayList<T>, ? extends Traversable<T>> collector() {
+        return this.<T>bsBuilder().collector();
+    }
+
+    @Override
+    protected <T> BitSet<T> empty() {
+        return this.<T>bsBuilder().empty();
+    }
+
+    @Override
+    protected boolean emptyShouldBeSingleton() {
+        return false;
+    }
+
+    @Override
+    protected <T> BitSet<T> of(T element) {
+        return this.<T>bsBuilder().of(element);
+    }
+
+    @SuppressWarnings("varargs")
+    @SafeVarargs
+    @Override
+    protected final <T> BitSet<T> of(T... elements) {
+        return this.<T>bsBuilder().of(elements);
+    }
+
+    @Override
+    protected boolean useIsEqualToInsteadOfIsSameAs() {
+        return true;
+    }
+
+    @Override
+    protected int getPeekNonNilPerformingAnAction() {
+        return 1;
+    }
+
+    @Override
+    protected <T> BitSet<T> ofAll(Iterable<? extends T> elements) {
+        return this.<T>bsBuilder().ofAll(elements);
+    }
+
+    @Override
+    protected BitSet<Boolean> ofAll(boolean[] array) {
+        return BitSet.ofAll(array);
+    }
+
+    @Override
+    protected BitSet<Byte> ofAll(byte[] array) {
+        return BitSet.ofAll(array);
+    }
+
+    @Override
+    protected BitSet<Character> ofAll(char[] array) {
+        return BitSet.ofAll(array);
+    }
+
+    @Override
+    protected BitSet<Double> ofAll(double[] array) {
+        return this.<Double>bsBuilder().ofAll(Iterator.ofAll(array));
+    }
+
+    @Override
+    protected BitSet<Float> ofAll(float[] array) {
+        return this.<Float>bsBuilder().ofAll(Iterator.ofAll(array));
+    }
+
+    @Override
+    protected BitSet<Integer> ofAll(int[] array) {
+        return BitSet.ofAll(array);
+    }
+
+    @Override
+    protected BitSet<Long> ofAll(long[] array) {
+        return BitSet.ofAll(array);
+    }
+
+    @Override
+    protected BitSet<Short> ofAll(short[] array) {
+        return BitSet.ofAll(array);
+    }
+
+    @Override
+    protected <T> BitSet<T> tabulate(int n, Function<? super Integer, ? extends T> f) {
+        return this.<T>bsBuilder().tabulate(n, f);
+    }
+
+    @Override
+    protected <T> BitSet<T> fill(int n, Supplier<? extends T> s) {
+        return this.<T>bsBuilder().fill(n, s);
+    }
+
+    @Override
+    protected BitSet<Character> range(char from, char toExclusive) {
+        return BitSet.range(from, toExclusive);
+    }
+
+    @Override
+    protected BitSet<Character> rangeBy(char from, char toExclusive, int step) {
+        return BitSet.rangeBy(from, toExclusive, step);
+    }
+
+    @Override
+    protected BitSet<Double> rangeBy(double from, double toExclusive, double step) {
+        return this.<Double>bsBuilder().ofAll(Iterator.rangeBy(from, toExclusive, step));
+    }
+
+    private static boolean badRange(int a, int b) {
```

Comment:
could you please prefix it with `is`, as I thought quickly that this method will generate illegal ranges :)